Allow Delete eviction policy on Azure VMs

allenhumphreys opened the pull request:
Per the [REST docs](https://docs.microsoft.com/en-us/rest/api/compute/virtual-machines/create-or-update#virtualmachineevictionpolicytypes), Delete has been allowed since API version `2019-03-01`.

I've confirmed this works (for linux) with a locally built provider, not sure if there are any tests that are affected.

Delete is a requirement of using ephemeral OS, e.g. this thing:

```hcl2
diff_disk_settings {
  option = "Local"
}
```

tombuildsstuff reviewed (changes requested):
hey @allenhumphreys 

Thanks for this PR - taking a look through whilst this looks good we'll want to add a test covering these ([example](https://github.com/hashicorp/terraform-provider-azurerm/blob/771c254f3b8bf73d8f3471ebeb81921df14f1d7c/internal/services/compute/linux_virtual_machine_scale_set_disk_os_resource_test.go#L86-L99) and [associated config](https://github.com/hashicorp/terraform-provider-azurerm/blob/771c254f3b8bf73d8f3471ebeb81921df14f1d7c/internal/services/compute/linux_virtual_machine_scale_set_disk_os_resource_test.go#L440)) and [update the documentation](https://github.com/hashicorp/terraform-provider-azurerm/blob/771c254f3b8bf73d8f3471ebeb81921df14f1d7c/website/docs/r/linux_virtual_machine_scale_set.html.markdown?plain=1#L159) (to include the new values) - but this otherwise LGTM 👍 

Thanks!

allenhumphreys commented:
@tombuildsstuff I've updated the documentation and added acceptance tests. I wasn't 100% sure about the test names. I took a stab at them.

Scenario:
- Ephemeral OS Disk
- Spot priority
- Delete eviction policy

The `Delete` eviction policy is required by the first 2 conditions. The API will reject a `Deallocate` policy with `"Standalone VMs with ephemeral OS disk support only \"Delete\" evictionPolicy setting when the VM is Spot. For more information, see http://aka.ms/AzureSpot/errormessages."` This differs from scale sets, and could potentially be validated in the provider.


Screenshots showing correct priority/eviction policy during Acc test runs

<img width="500" alt="Screen Shot 2022-06-14 at 10 28 53 AM" src="https://user-images.githubusercontent.com/8225090/173607468-48acad54-8f97-4c68-8686-50db5d854788.png">
<img width="500" alt="Screen Shot 2022-06-14 at 10 42 06 AM" src="https://user-images.githubusercontent.com/8225090/173607476-167cfe7d-f02e-4b8e-8ac0-574bc6993e02.png">

katbyte reviewed (approved):
Thanks @allenhumphreys - LGTM 🍍
